evdns.c

```diff
@@ -742,9 +742,37 @@ nameserver_failed(struct nameserver *const ns, const char *msg) {
 	ns->state = 0;
 	ns->failed_times = 1;
 
-	disconnect_and_free_connection(ns->connection);
-	ns->connection = NULL;
+	if (ns->connection) {
+		disconnect_and_free_connection(ns->connection);
+		ns->connection = NULL;
+	} else if (err == ENOTCONN) {
+		/* XXX: on iOS, when app become background, socket may become unusable.
+		 * But when it will be also readable for kqueue/poll.
+		 * It lead infinite loop of recvfrom -> error -> recvfrom -> error.
+		 * we recreate socket in the ENOTCONN error to break this infinite loop.
```

Comment:
Sorry, message is not clear enough for me, I'm suggesting you to just copy description from the linked issue

Reply:
copied

Reply:
Have you pushed the changes? If you want to, I can finish by myself (to reduce latency for interaction) or I can wait, it's up to you

Reply:
I'm sorry for forgeting push.
pushed currently